tests/integration/scenarios/internal_http_server/tests/test_internal_http_server_exec_deadlock_001.py

```
@@ -0,0 +1,90 @@
+            ("/api/v1/metrics", "output"),
+            ("/api/v1/metrics/prometheus", "fluentbit_uptime"),
+            ("/api/v1/storage", "chunks"),
+            ("/api/v2/metrics", "fluentbit_uptime"),
+            ("/api/v2/metrics/prometheus", "fluentbit_uptime"),
+        ]
+
+        for path, pattern in endpoints:
+            result = service.service.wait_for_condition(
+                lambda: (
+                    response
+                    if response["status_code"] == 200 and pattern in response["body"]
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `pattern`

(B023)